src/devtoolsPanel.ts

```diff
@@ -304,6 +306,61 @@ export class DevToolsPanel {
             return;
         }
 
+        const uri = await this.parseUrlToUri(url);
+        if (uri) {
+            void this.openEditorFromUri(uri, line, column);
+        } else {
+            void vscode.window.showErrorMessage(`Could not open document. No workspace mapping was found for '${url}'.`);
+        }
+    }
+
+    private async onSocketCssMirrorContent(message: string) {
+        if (!vscode.workspace.getConfiguration(SETTINGS_STORE_NAME).get('mirrorEdits')) {
+            return;
+        }
+
+        // Parse message and open the requested file
+        const { url, newContent } = JSON.parse(message) as ICssMirrorContentData;
+
+        const uri = await this.parseUrlToUri(url);
+
+        // Finally open and edit the document if it exists
+        if (uri) {
+            const textEditor = await this.openEditorFromUri(uri);
+            if (textEditor) {
+                const fullRange = this.getDocumentFullRange(textEditor);
+                void textEditor.edit(editBuilder => {
+                    editBuilder.replace(fullRange, newContent);
+                });
+            }
+        } else {
+            void vscode.window.showErrorMessage(`Could not mirror css changes to document. No workspace mapping was found for '${url}'.`);
+        }
+    }
+
+    private async openEditorFromUri(uri: vscode.Uri, line?: number, column?: number): Promise<vscode.TextEditor | undefined> {
+        try {
+            const doc = await vscode.workspace.openTextDocument(uri);
+            const viewColumn = this.panel.viewColumn === vscode.ViewColumn.One ? vscode.ViewColumn.Beside : vscode.ViewColumn.One;
+            const selection = line !== undefined && column !== undefined ? new vscode.Range(line, column, line, column) : undefined;
+            return await vscode.window.showTextDocument(doc, { preserveFocus: true, viewColumn, selection });
+        } catch (e) {
+            await ErrorReporter.showErrorDialog({
+                errorCode: ErrorCodes.Error,
+                title: 'Error while opening file in editor.',
+                message: e,
+            });
+        }
+    }
+
+    private getDocumentFullRange(textEditor: vscode.TextEditor): vscode.Range {
+        const firstLine = textEditor.document.lineAt(0);
+        const lastLine = textEditor.document.lineAt(textEditor.document.lineCount - 1);
```

Comment:
is there any case where this might end up calling lineAt(-1)? e.g if the user deletes the whole content of the file and then reopens it? (not sure if the scenario is possible just thought about this corner case)

Reply:
I think we should be fine here.  Tried breaking it a bunch of different ways, but in the end a blank file still starts at line 1.